src/GovUk.Education.ExploreEducationStatistics.Content.Search.FunctionApp.Tests/Functions/OnReleaseVersionPublished/OnReleaseVersionPublishedFunctionTests.cs

```diff
@@ -34,21 +31,78 @@ public async Task GivenEvent_WhenPayloadContainsPublicationSlug_ThenRefreshSearc
         var response = await sut.OnReleaseVersionPublished(eventGridEvent, new FunctionContextMockBuilder().Build());
         
         // ASSERT
-        var actual = Assert.Single(response);
+        Assert.NotNull(response);
+        var actual = Assert.Single(response.RefreshSearchableDocumentMessages);
         Assert.NotNull(actual);
         Assert.Equal(payload.PublicationSlug, actual.PublicationSlug);
     }
     
     [Theory]
     [MemberData(nameof(TheoryDatas.Blank.Strings), MemberType = typeof(TheoryDatas.Blank))]
-    public async Task GivenEvent_WhenPayloadDoesNotContainPublicationSlug_ThenNothingIsReturned(string? blankPublicationSlug)
+    public async Task GivenEvent_WhenPayloadDoesNotContainPublicationSlug_ThenEmptyIsReturned(string? blankPublicationSlug)
     {
         // ARRANGE
-        var payload = new ReleaseVersionPublishedEventDto
-        {
-            PublicationSlug = blankPublicationSlug,
-        };
+        var payload = NewReleaseVersionPublishedEvents.NoPublicationSlug();
+        
+        var eventGridEvent = new EventGridEventBuilder()
+            .WithPayload(payload)
+            .Build();
+
+        var sut = GetSut();
+        
+        // ACT
+        var response = await sut.OnReleaseVersionPublished(eventGridEvent, new FunctionContextMockBuilder().Build());
+        
+        // ASSERT
+        Assert.Equal(OnReleaseVersionPublishedOutput.Empty, response);
+    }
+
+    [Fact]
+    public async Task GivenEvent_WhenPublishedReleaseVersionIsNotTheLatest_ThenEmptyIsReturned()
+    {
+        // ARRANGE
+        var payload = NewReleaseVersionPublishedEvents.NewlyPublishedIsNotLatest;
+        
+        var eventGridEvent = new EventGridEventBuilder()
+            .WithPayload(payload)
+            .Build();
+
+        var sut = GetSut();
+        
+        // ACT
+        var response = await sut.OnReleaseVersionPublished(eventGridEvent, new FunctionContextMockBuilder().Build());
+        
+        // ASSERT
+        Assert.Equal(OnReleaseVersionPublishedOutput.Empty, response);
+    }
+    
+    // New release, then ensure old searchable doc is removed
+    [Fact]
+    public async Task GivenEvent_WhenPublishedReleaseVersionIsADifferentRelease_ThenPreviousReleaseSearchableDocumentIsRemoved()
+    {
+        // ARRANGE
+        var payload = NewReleaseVersionPublishedEvents.NewlyPublishedIsLatestButDifferentRelease;
+        
+        var eventGridEvent = new EventGridEventBuilder()
+            .WithPayload(payload)
+            .Build();
 
+        var sut = GetSut();
+        
+        // ACT
+        var response = await sut.OnReleaseVersionPublished(eventGridEvent, new FunctionContextMockBuilder().Build());
+        
+        // ASSERT
+        Assert.Equal([new RemoveSearchableDocumentDto{ ReleaseId = payload.PreviousLatestReleaseId }], response.RemoveSearchableDocuments);
+    }
+    
+    // Different release version for the same release, then no document is removed - it will be overwritten
```

Comment:
Up to you, but I think you could reword this slightly to explain why we asserting, in-lined to be just above the assert.